components/global/product_switcher_menu/product_switcher_menu.test.tsx

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import {shallowWithIntl} from 'tests/helpers/intl-test-helper';
+import {TestHelper} from 'utils/test_helper';
+
+import ProductSwitcherMenu, {Props as ProductSwitcherMenuProps} from './product_switcher_menu';
+
+describe('components/global/product_switcher_menu', () => {
+    // Neccessary for components enhanced by HOCs due to issue with enzyme.
+    // See https://github.com/enzymejs/enzyme/issues/539
+    const getMenuWrapper = (props: ProductSwitcherMenuProps) => {
+        const wrapper = shallowWithIntl(<ProductSwitcherMenu {...props}/>);
+        return wrapper.find('MenuGroup').shallow();
+    };
+
+    const user = TestHelper.getUserMock({
+        id: 'test-user-id',
+        username: 'username',
+    });
+
+    const defaultProps: ProductSwitcherMenuProps = {
+        isMobile: false,
+        id: '',
+        teamId: '',
+        teamName: '',
+        siteName: '',
+        currentUser: user,
+        appDownloadLink: 'test–link',
+        isMessaging: true,
+        enableCommands: false,
+        enableIncomingWebhooks: false,
+        enableOAuthServiceProvider: false,
+        enableOutgoingWebhooks: false,
+        canManageSystemBots: false,
+        canManageIntegrations: true,
+        enablePluginMarketplace: false,
+        onClick: () => jest.fn,
+    };
+
+    test('should match snapshot with id', () => {
```

Comment:
nit and my personal preference: 
the test case naming is lacking expressiveness when using `should`.

IMO this would be better as: `matches snapshot with id`

Reply:
Good point, I would stick to the convention in Mattermost for now and maybe we can discuss naming on a later stage.